tests/check_framework/tests.py

```diff
@@ -305,3 +305,30 @@ class ModelWithDescriptorCalledCheck(models.Model):
 class ChecksRunDuringTests(SimpleTestCase):
     def test_registered_check_did_run(self):
         self.assertTrue(my_check.did_run)
+
+class CheckDuplicateDBTableTests(SimpleTestCase):
+
+    @isolate_apps('check_framework', kwarg_name='apps')
+    @override_system_checks([checks.model_checks.check_all_models])
+    def test_model_check_method_not_shadowed(self, apps):
+
+        class ModelWithDBTableDefined(models.Model):
+
+            class Meta:
+                db_table = 'test_table'
+
+        class AnotherModelWithDBTableDefined(models.Model):
+
+            class Meta:
+                db_table = 'test_table'
+
+        errors = checks.run_checks(app_configs=apps.get_app_configs())
+        expected = [
+            Error(
+                "db_table 'test_table' was declared in multiple models: "
+                "'ModelWithDBTableDefined, AnotherModelWithDBTableDefined'",
+                obj=AnotherModelWithDBTableDefined,
+                id='models.E020'
+            ),
+        ]
+        self.assertEqual(errors, expected)
```

Comment:
I'd add two extra tests:

1. One for cross-app collisions.
2. One for unmanaged model collisions (check should not report any errors).

Reply:
I've added these tests and PR is ready for the review